### PR TITLE
Add enumeration of common dnsmasq warnings

### LINKED ISSUE #600: Add documentation for common dnsmasq warnings
In the upcoming Pi-hole version, the diagnosis system will also show `dnsmasq` warnings. To reduce user confusion and support requests, we should add common dnsmasq warnings and their interpretation to the documentation

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Fix #600 

**How does this PR accomplish the above?:**

Add enumeration of common `dnsmasq` warnings

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

File: docs/ftldns/dnsmasq_warn.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+As the warning says. No action is performed in this case.
+
+## No IPv4 address found for `HOSTNAME`
+
+Lookup for an A record in the cache returned no result.
+
+## `HOSTNAME` is a CNAME, not giving it to the DHCP lease of `ADDRESS`
+
+A hostname claimed by a DHCP client is a known CNAME. `dnsmasq` des
```
Comment (yubiuser) [inline docs/ftldns/dnsmasq_warn.md:23]: Line ends in nothing.

File: docs/ftldns/dnsmasq_warn.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+This warning highlights that one of the two addresses is outside of the configured subnet mask. As a consequence, not all addresses may be reachable from configured hosts leading to unexpected behavior on the clients. Make your DHCP settings consistent.
+
+## Ignoring duplicate dhcp-option `OPTNUM`
+
+DHCP options specified more than once are ignored. The corresponding option ID is given by `OPTNUM`
+
+## `HOSTNAME` has more than one address in hostsfile, using `ADDRESS` for DHCP
+
+Some people like to keep all static IP addresses in `/etc/hosts`. `dnsmasq` goes through `/etc/hosts` and sets static addresses for any DHCP config records which don't have an address and whose name matches qhere `dnsmasq` maintains the invariant that any IP address can appear in at least one DHCP host.
```
Comment (yubiuser) [inline docs/ftldns/dnsmasq_warn.md:51]:
```suggestion
Some people like to keep all static IP addresses in `/etc/hosts`. `dnsmasq` goes through `/etc/hosts` and sets static addresses for any DHCP config records which don't have an address and whose name matches here. `dnsmasq` maintains the invariant that any IP address can appear in at least one DHCP host.
```

File: docs/ftldns/dnsmasq_warn.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+## duplicate IP address `ADDRESS` (`HOSTNAME`) in dhcp-config directive
+
+As the warning says.
+
+## cache size greater than 10000 may cause performance issues, and is unlikely to be useful.
+
+This causes the cache to consume a lot on memory and slows down cache lookups. As expiring cache entries naturally make room for new records, a large cache does not give any advantages beyond a certain level. This level is typically not very high. Try reducing the cache. Watch out for cache-evictions. If they are zweo over a long time, your cache is larger than what you need.
```
Comment (yubiuser) [inline docs/ftldns/dnsmasq_warn.md:59]:
```suggestion
This causes the cache to consume a lot on memory and slows down cache lookups. As expiring cache entries naturally make room for new records, a large cache does not give any advantages beyond a certain level. This level is typically not very high. Try reducing the cache. Watch out for cache-evictions. If they are zero over a long time, your cache is larger than what you need.
```

File: docs/ftldns/dnsmasq_warn.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+## warning: no upstream servers configured
+
+Only local names can be answered as no `server=...` lines are defined.
+
+## warning: `TFTP_PREFIX` inaccessible
+
+The TFTP prefix (set by `tftp-prefix`) is inaccessible or not a directory.
+
+## warning: TFTP directory %s inaccessible
```
Comment (yubiuser) [inline docs/ftldns/dnsmasq_warn.md:85]:
```suggestion
## warning: `TFTP directory %s` inaccessible
```

File: docs/ftldns/dnsmasq_warn.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+Upstream at address `ADDRESS` is missing the `RA` (recursion available) bit. This warning is printed only once per server.
+
+## possible DNS-rebind attack detected: `NAME`
+
+`A` and `AAAA` answers are checked against possible rebind attacks when `stop-dns-rebind` is enabled. See `rebind-domain-ok=/domain/` for adding exceptions.
+
+## reducing DNS packet size for nameserver `ADDRESS` to `SAFE_PKTSZ`
+
+When receiving anwers from upstream only with a smaller maximum DNS packet size, `dnsmasq` remembers this decision and makes it permanent in the current session.
```
Comment (yubiuser) [inline docs/ftldns/dnsmasq_warn.md:127]:
```suggestion
When receiving answers from upstream only with a smaller maximum DNS packet size, `dnsmasq` remembers this decision and makes it permanent in the current session.
```

File: docs/ftldns/dnsmasq_warn.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+As the warning says. Check your log file for reasons of a prior refusal to hand out this lease. This warning is at most logged once every 10 minutes for a given address.
+
+## cannot send DHCP/BOOTP option `NUMBER`: no space left in packet
+
+Use less DHCP options as the space for options is limited and cannot be extended (RFC2131).
+
+## cannot send RFC3925 option: too many options for enterprise number `NUMBER`
+
+A maximum packet length of 250 bytes has to be ensured for `dhcp-option = vi-encap:13,17,.......` configurations.
```
Comment (yubiuser) [inline docs/ftldns/dnsmasq_warn.md:223]:
```suggestion
A maximum packet length of 250 bytes has to be ensured for `dhcp-option = vi-encap:13,17,...` configurations.
```

**Review by yubiuser — COMMENTED** [review]
In general I would use a smaller size fore each heading, `###` or even `####`.

Should we add a link to this doc to the message table?

**Comment by DL6ER** [comment]
> In general I would use a smaller size fore each heading, ### or even ####.

It does look nice at any level as the description should somehow be smaller than the title IMO. I will look at other options.

### `##`
two

### `###`
three

### `####`
four

### `#####`
five

**Comment by DL6ER** [comment]
Proposed alternative:

Screenshot from 2021-12-04 15-37-46

**Comment by DL6ER** [comment]
Another possible alternative would be:
Screenshot from 2021-12-04 15-46-55

**Comment by yubiuser** [comment]
I'll like the new look.
Let's fix the complains and get this merged :)

**Comment by DL6ER** [comment]
Local `npm test` didn't complain besides updated components, will look at this tomorrow.

File: docs/ftldns/dnsmasq_warn.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
+# Known `dnsmasq` warnings
+
+Warnings commonly seen in `dnsmasq`'s log file (`/var/log/pihole.log`) and the Pi-hole diagnosis system.
+
```
Comment (dschaper) [inline docs/ftldns/dnsmasq_warn.md:4]:
```suggestion
<!-- markdownlint-disable MD046 -->
```

**Review by yubiuser — APPROVED** [review]

**Pull request merged**